kernel/mellanox_ethernet/src/command_queue.rs

```diff
@@ -0,0 +1,740 @@
+    command_output_status:          Volatile<U32<BigEndian>>,
+    /// Syndrome on the command, valid only if status = 0x0
+    command_output_syndrome:        Volatile<U32<BigEndian>>,
+    /// First 4 bytes of command output data, valid only if status = 0x0
+    command_output_inline_data_0:   Volatile<U32<BigEndian>>,
+    /// Second 4 bytes of command output data, valid only if status = 0x0
+    command_output_inline_data_1:   Volatile<U32<BigEndian>>,
+    /// Pointer to output mailbox, upper 4 bytes
+    output_mailbox_pointer_h:       Volatile<U32<BigEndian>>,
+    /// Pointer to output mailbox, lower 4 bytes
+    output_mailbox_pointer_l:       Volatile<U32<BigEndian>>,
+    /// Output command length in bytes
+    output_length:                  Volatile<U32<BigEndian>>,
+    /// * Token: Token of the command, should have the same value in the command and the mailbox blocks. 
+    /// * Signature: 8 bit signature of the command queue entry.
+    /// * Status: command delivery status.
+    /// * Ownership: bit 0 of field. When set, indicates that HW owns the command entry.
+    token_signature_status_own:     Volatile<U32<BigEndian>>
+}
+
+const_assert_eq!(core::mem::size_of::<CommandQueueEntry>(), 64);
+
+impl fmt::Debug for CommandQueueEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "CQE:type of transport: {:#X} \n", self.type_of_transport.read().get())?;
+        write!(f, "CQE:input length: {} \n", self.input_length.read().get())?;
+        write!(f, "CQE:input_mailbox_ptr_h: {:#X} \n", self.input_mailbox_pointer_h.read().get())?;
+        write!(f, "CQE:input_mailbox_ptr_l: {:#X} \n", self.input_mailbox_pointer_l.read().get())?;
+        write!(f, "CQE:command_input_opcode: {:#X} \n", self.command_input_opcode.read().get())?;
+        write!(f, "CQE:command_input_opmod: {:#X} \n", self.command_input_opmod.read().get())?;
+        write!(f, "CQE:command_input_inline_data_0: {:#X} \n", self.command_input_inline_data_0.read().get())?;
+        write!(f, "CQE:command_input_inline_data_1: {:#X} \n", self.command_input_inline_data_1.read().get())?;
+        write!(f, "CQE:command_output_status: {:#X} \n", self.command_output_status.read().get())?;
+        write!(f, "CQE:command_output_syndrome: {:#X} \n", self.command_output_syndrome.read().get())?;
+        write!(f, "CQE:command_output_inline_data_0: {:#X} \n", self.command_output_inline_data_0.read().get())?;
+        write!(f, "CQE:command_output_inline_data_1: {:#X} \n", self.command_output_inline_data_1.read().get())?;
+        write!(f, "CQE:output_mailbox_pointer_h: {:#X} \n", self.output_mailbox_pointer_h.read().get())?;
+        write!(f, "CQE:output_mailbox_pointer_l: {:#X} \n", self.output_mailbox_pointer_l.read().get())?;
+        write!(f, "CQE:output_length: {} \n", self.output_length.read().get())?;
+        write!(f, "CQE:token_signature_status_own: {:#X} \n", self.token_signature_status_own.read().get())
```

Comment:
This technically works, but isn't flexible -- it doesn't support Rust's various fmt modifiers like `{:?}` vs `{:#?}` vs `{:#X?}`, etc. 

The proper way to do it is to use the fmt debug helper functions, like `debug_struct()` and the `field()` functions. Search the code base for `.debug_struct` to see examples of this. 

links: https://doc.rust-lang.org/core/fmt/struct.Formatter.html#method.debug_struct